### PR TITLE
chore(IDX): bazel --profile=profile.json

### PR DESCRIPTION
**Pull request by marko-k0**
Adding `--profile=profile.json` to save a build profile (with critical path info). Changing retention to 30 days.
Build profile data example:
image

### CODE REVIEW

**Comment by nmattia** [comment]
@marko-k0 what is written to that file and what is it used for?

**Comment by marko-k0** [comment]
> @marko-k0 what is written to that file and what is it used for?

As said in description "Adding --profile=profile.json to save the critical path of the build". It will be saved to superset along with pb build time data. You can look at last workflow and inspect json artifact.

**Comment by marko-k0** [comment]
> @marko-k0 what is written to that file and what is it used for?

I've updated description. Please let me know if there's still something unclear. You can look into example json by downloading example artifact [here](https://github.com/dfinity/ic/actions/runs/10410770713/artifacts/1817672775).

File: .github/workflows-source/ci-main.yml [hunk 1]
```diff
@@ -113,19 +113,22 @@ jobs:
         uses: actions/upload-artifact@v4
         with:
           name: bazel-targets
-          retention-days: 1
+          retention-days: 30
```
Comment (cgundy) [inline .github/workflows-source/ci-main.yml:116]: Hey I think we'll need to be a little careful here, as we don't have unlimited storage. Separately, isn't the build profile based on data in ceph so we'd want to actually increase the retention days there?

Comment (marko-k0) [reply .github/workflows-source/ci-main.yml:116]: I'll change it to 14 days and we can see latter on how much roughly that contributes. Leaving retention to 1 day in CEPH is OK - if we see non-deterministic build, we can pull artifacts from CEPH immediately and investigate. I've aligned it here to be the same as for `bazel-bep.pb` and `profile.json` as the idea is to also save the targets that were modified in the PR along with all the build stats.

**Review by cgundy — APPROVED** [review]

**Pull request merged**